identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala

```diff
@@ -0,0 +1,43 @@
+package com.gu.identity.integration.test.pages
+
+import com.gu.integration.test.pages.common.ParentPage
+import org.openqa.selenium.support.ui.{WebDriverWait, ExpectedConditions}
+import org.openqa.selenium.{By, WebDriver, WebElement}
+
+class FaceBookAuthDialog(implicit driver: WebDriver) extends ParentPage {
+  private def confirmButton: WebElement = {
+    val xpath = "id('platformDialogForm')//button[@name='__CONFIRM__']"
```

Comment:
This feels quite risky to use this kind of XPath traversal which can break easily. Have you looked at a programmatic way of authenticating? I think [Koala](https://github.com/arsduo/koala) can do it though it is Ruby maybe you could use something similar?